src/tools/clang.lua

```diff
@@ -350,6 +350,11 @@
 	}
 
 	function clang.gettoolname(cfg, tool)
+		-- Check toolsetpaths first
+		if cfg.toolsetpaths and cfg.toolsetpaths[cfg.toolset] and cfg.toolsetpaths[cfg.toolset][tool] then
+			return cfg.toolsetpaths[cfg.toolset][tool]
+		end
```

Comment:
Would be fine to have something like:
`function toolset.gettoolname(toolset, config, tool)` to factorize that code.
but probably not possible as it is already used by exporters :-/